### PR TITLE
feat: add Dot component

### PR DESCRIPTION
**Pull request by dhavelock**
+ Dot component
+ util script to create component directory + files

### CODE REVIEW

File: scripts/new.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (alamorre) [inline scripts/new.sh:1]: Dirty 🤤🤤🤤

I even have template user snippets for each one I'll beef this up a bit

File: src/Dot/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { stringToColor } from '../util/colorMapping';
+import { Props } from './props';
+import { styles } from './styles';
+
+export const Dot = ({
+  avatarUrl = undefined,
+  username = '',
+  style = {},
+  visible = false
```
Comment (alamorre) [inline src/Dot/index.tsx:10]: Maybe we should default this to true?

Would be weird if somebody implements it and nothing shows up haha

File: src/Dot/props.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import CSS from 'csstype';
+
+export interface Props {
+  avatarUrl?: string,
+  username?: string,
+  style?: CSS.Properties,
+  visible: boolean
```
Comment (alamorre) [inline src/Dot/props.tsx:7]: Yeah I'd make this not required.

When people setup a dot it should just show unless o.t.w specified

File: src/Dot/stories.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const Template: Story<Props> = ({ avatarUrl, username, style, visible }) => (
+  <Dot avatarUrl={avatarUrl} username={username} style={style} visible={visible} />
+);
+
+export const Default = Template.bind({});
+Default.args = {
+  avatarUrl: 'https://chat-engine-assets.s3.amazonaws.com/tutorials/my-face-min.png',
+  username: 'Dylan',
+  style: { float: 'right', marginLeft: '4px' },
+  visible: true
```
Comment (alamorre) [inline src/Dot/stories.tsx:27]: Then yeah take this out 👍

File: src/util/colorMapping.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+const colors = [
```
Comment (alamorre) [inline src/util/colorMapping.ts:1]: I think `util` is a good name - good stuff!

File: src/Dot/stories.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+export const NoAvatar = Template.bind({});
```
Comment (alamorre) [inline src/Dot/stories.tsx:30]: Can we call this NoAvatarUrl again to be as specific as the prop renaming?

File: src/Dot/stories.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+NoAvatar.args = {
+  avatarUrl: undefined,
+  username: 'Dylan',
+  style: { float: 'right', marginLeft: '4px' },
+  visible: true
+};
+
+export const TheirMessage = Template.bind({});
```
Comment (alamorre) [inline src/Dot/stories.tsx:38]: I don't see any "their" logic in these components? Is there a difference / is this example educational?

**Comment by alamorre** [comment]
I see like 2-3 small todos after that I'll totally approve!

File: scripts/new.sh (hunk 1 above)
Comment (dhavelock) [reply scripts/new.sh:1]: 💯

File: src/Dot/stories.tsx (hunk 7 above)
Comment (dhavelock) [reply src/Dot/stories.tsx:38]: The point was to show how they would use it but I guess we will really just be passing this to another component so kinda useless- I'll take out

**Review by alamorre — APPROVED** [review]
Goochy

**Pull request merged**